libkv.go

```diff
@@ -30,29 +33,98 @@ func NewLibKV(url *url.URL) *LibKV {
 
 	sourceType = ""
 	client = ""
+	options := store.Config{}
 
 	if url.Scheme == "consul" {
 		consul.Register()
 		sourceType = store.CONSUL
 		client = env.Getenv("CONSUL_HTTP_ADDR", "localhost:8500")
+		if timeout := env.Getenv("CONSUL_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("consul timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("CONSUL_HTTP_SSL", ""); ssl != "" {
```

Comment:
@stuart-c hmm... on the consul docs, `CONSUL_HTTP_ADDR` doesn't refer to a remote Consul agent, but the local Consul agent that you're starting. So maybe we're overloading it in a bit of a confusing way.

Also, from the docs:
```
This can also be specified via the CONSUL_HTTP_ADDR environment
     variable. The default value is http://127.0.0.1:8500. The scheme
     can also be set to HTTPS by setting the environment variable
     CONSUL_HTTP_SSL=true.
```

In that example (in https://www.consul.io/docs/commands/index.html) a scheme is used, but in a further example for `CONSUL_HTTP_ADDR` directly, the value is scheme-less.

I'd recommend changing to `CONSUL_URL` to reduce the confusion (and potential conflicts), and using the scheme to reflect TLS or not.